From 354da027ea0f38f82646550189ed048c75a96485 Mon Sep 17 00:00:00 2001
From: Yazan Shhady <yazan.shhady@solid-run.com>
Date: Mon, 10 Apr 2023 18:21:56 +0300
Subject: [PATCH] add eeprom and sensors support to rzg2lc-compact dts

- Add Eeprom CAT24AA01TDI-GT3 1-kb (128 B x 8) suport [I2C0-0x50]
- Add Accelerometer I2C sensor support [I2C0-0x1d]
- Add ambient light I2C Sensor support [I2C0-0x44]
---
 .../arm64/boot/dts/renesas/rzg2lc-compact.dts | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/rzg2lc-compact.dts b/arch/arm64/boot/dts/renesas/rzg2lc-compact.dts
index 202619fadc2e..8d160bc60755 100644
--- a/arch/arm64/boot/dts/renesas/rzg2lc-compact.dts
+++ b/arch/arm64/boot/dts/renesas/rzg2lc-compact.dts
@@ -359,6 +359,25 @@ &i2c0 {
 	pinctrl-0 = <&i2c0_pins>;
 	pinctrl-names = "default";
 	status = "okay";
+
+  /* EEPROM CAT24AA01TDI-GT3 1-kb (128 B x 8)  */
+  eeprom: eeprom@50 {
+    compatible = "atmel,24c02";
+    reg = <0x50>;
+    pagesize = <16>;
+  };
+
+  accelerometer@1d {
+    compatible = "adi,adxl345";
+    reg = <0x1d>;
+    //interrupt-parent = <&gpio1>; //I/O Expander IO0_6
+    interrupts = <9 IRQ_TYPE_LEVEL_HIGH>;
+  };
+
+  ambient_light_sensor: isl29023@44 {
+    compatible = "isil,isl29023";
+    reg = <0x44>;
+  };
 };
 
 &i2c1 {
-- 
2.25.1

